draft-irtf-cfrg-hpke.md

```diff
@@ -1451,16 +1454,20 @@ achieved by the different prefix-free label parameters in the calls to
 domains of all `Extract()` and `Expand()` invocations. It also justifies modeling
 them as independent functions even if instantiated by the same KDF.
 
-Future KEM instantiations MUST ensure that all internal invocations of
-`Extract()` and `Expand()` can be modeled as functions independent from the
-invocations of `Extract()` and `Expand()` in the remainder of HPKE. One way to
-ensure this is by using an equal or similar prefixing scheme with
-an identifier different from "HPKE-v1". Particular attention needs to
+Future KEM instantiations MUST ensure, should `Extract()` and
+`Expand()` be used internally, that they can be modeled as functions
+independent from the invocations of `Extract()` and `Expand()` in the
+remainder of HPKE. One way to ensure this is by using `LabeledExtract()`
+and `LabeledExpand()` with label parameters that are prefix-free among
+each other and prefix-free with respect to the labels used in the remainder
```

Comment:
Are you saying that prefix-free applies to all labels passed to `LabeledE...()`?  That is, a label for one function cannot be a prefix for the other function?  I don't think that is necessary.  If you said that, is there a risk that the choice of length for `LabeledExpand()` causes this property to be violated?

Reply:
It's also not totally clear to me what "prefix-free" means here.  I think you mean "no label is a prefix of another within the set"?  As opposed to something like, "no two labels share a non-empty common prefix".  Assuming that, an extra parenthetical of the form "(A set is "prefix-free" if ...)" would be helpful here.

Reply:
@martinthomson Thanks, you are right. The advice was overly restrictive.

Because of the shift of two bytes at the beginning of the label, the domains of Extract and Expand are already separated (`KE` is different from `HP`):
```
HPKE-v1...
LLHPKE-v1...
```

Because of the start of the `suite_id`, the domains of Extract and Expand in the KEM vs the remainder of HPKE are already separated:
```
KEMID
HPKEIDIDID
```

So indeed, the labels for one function could be the prefix of the one of another function, and it would not invalidate the security theorems.

I rephrased the first and second paragraph accordingly.

@bifurcation I added a definition of prefix-free following your suggestion, thanks.